src/wg/color/rgb/package.d

```diff
@@ -2,3 +2,136 @@ module wg.color.rgb;
 
 public import wg.color.rgb.colorspace;
 public import wg.color.rgb.format;
+
+import wg.util.normint;
+
+/**
+ * RGB colour type.
+ */
+struct RGB(string format)
+{
+    // mixin the members
+    mixin Members!(typeof(this));
+
+    // make the format data available
+    static immutable Format = parseRGBFormat(format);
+}
+///
+unittest
+{
+    RGB!"rgba_8_s8_i8_u8" pixel;
+
+    static assert([ __traits(allMembers, typeof(pixel)) ] == [ "r", "g", "b", "a", "Format" ]);
+    static assert(is(typeof(pixel.r) == NormalizedInt!ubyte));
+    static assert(is(typeof(pixel.g) == NormalizedInt!byte));
+    static assert(is(typeof(pixel.b) == byte));
+    static assert(is(typeof(pixel.a) == ubyte));
+
+    RGB!"la_f32_f32" pixel2;
```

Comment:
I feel like I could support an abbreviation, where if all components are the same, then just one needs to be stated...?

Reply:
why not, but isn't it pre-optimization?

Reply:
yeah, but i'm also thinking about user-facing usability, and part of that is whether this form of description is a good idea in general.
I think I'm feeling pretty good about it.